pkg/storegateway/bucket_stores.go

```diff
@@ -151,11 +152,31 @@ func (u *BucketStores) InitialSync(ctx context.Context) error {
 
 // SyncBlocks synchronizes the stores state with the Bucket store for every user.
 func (u *BucketStores) SyncBlocks(ctx context.Context) error {
-	return u.syncUsersBlocks(ctx, func(ctx context.Context, s *store.BucketStore) error {
+	return u.syncUsersBlocksWithRetries(ctx, func(ctx context.Context, s *store.BucketStore) error {
 		return s.SyncBlocks(ctx)
 	})
 }
 
+func (u *BucketStores) syncUsersBlocksWithRetries(ctx context.Context, f func(context.Context, *store.BucketStore) error) error {
+	retries := util.NewBackoff(ctx, util.BackoffConfig{
+		MinBackoff: 100 * time.Millisecond,
+		MaxBackoff: 10 * time.Second,
+		MaxRetries: 3,
```

Comment:
Nit: Given that `maxRetries` is 3, we can never reach max backoff of 10 seconds. :)

Reply:
Ah ok I originally stole that from [here](https://github.com/cortexproject/cortex/blob/f6298030538d5582e879eb2074f94f3bf5e116d3/pkg/util/backoff.go#L20-L22) and just modified it along the way. Is a 1-second `MaxBackoff` ok?

Reply:
It's ok to keep it as is. If you choose to use 1s as MaxBackoff, that's fine too. I think it's more important to do retry here, than how much time it waits.